src/osd/PeeringState.cc

```diff
@@ -2147,7 +2147,7 @@ PeeringState::WaitLocalRecoveryReserved::WaitLocalRecoveryReserved(my_context ct
 
   // Make sure all nodes that part of the recovery aren't full
   if (!ps->cct->_conf->osd_debug_skip_full_check_in_recovery &&
-      ps->pg->osd->check_osdmap_full(ps->acting_recovery_backfill)) {
+      ps->get_osdmap()->check_full(ps->acting_recovery_backfill)) {
```

Comment:
This seems fine to me since if a full peer isn't marked full in the current map the PG has, the remote reservation will end up being rejected

Reply:
For the moment, I chose to try to preserve the existing logic -- I'm inclined to defer making that sort of improvement until after this series.

Reply:
Also, avoiding the primary local reservation may well be worth it anyway in the case that the cluster is filling up like that.